lib/sdf-server/src/service/v2/variant/generate_aws_asset_schema.rs

```diff
@@ -0,0 +1,94 @@
+use axum::extract::{Host, OriginalUri, Path, Query};
+use dal::{
+    schema::variant::authoring::VariantAuthoringClient, ChangeSet, ChangeSetId, SchemaVariant,
+    SchemaVariantId, WorkspacePk, WsEvent,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    extract::{AccessBuilder, AssetSprayer, HandlerContext, PosthogClient},
+    service::force_change_set_response::ForceChangeSetResponse,
+    track,
+};
+
+use super::SchemaVariantsAPIResult;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct AwsCommand {
+    pub command: String,
+    pub subcommand: String,
+}
+
+pub async fn generate_aws_asset_schema(
+    HandlerContext(builder): HandlerContext,
+    AssetSprayer(asset_sprayer): AssetSprayer,
+    AccessBuilder(access_builder): AccessBuilder,
+    PosthogClient(posthog_client): PosthogClient,
+    OriginalUri(original_uri): OriginalUri,
+    Host(host_name): Host,
+    Path((_workspace_pk, change_set_id, schema_variant_id)): Path<(
+        WorkspacePk,
+        ChangeSetId,
+        SchemaVariantId,
+    )>,
+    Query(aws_command): Query<AwsCommand>,
+) -> SchemaVariantsAPIResult<ForceChangeSetResponse<()>> {
+    let mut ctx = builder
+        .build(access_builder.build(change_set_id.into()))
+        .await?;
+
+    // Generate the code
+    let code = asset_sprayer
+        .aws_asset_schema(&aws_command.command, &aws_command.subcommand)
+        .await?;
+
+    // Update the function
+    let force_change_set_id = ChangeSet::force_new(&mut ctx).await?;
+
+    let schema_variant = SchemaVariant::get_by_id_or_error(&ctx, schema_variant_id).await?;
+    let schema_id = SchemaVariant::schema_id_for_schema_variant_id(&ctx, schema_variant_id).await?;
+    let variant = schema_variant.into_frontend_type(&ctx, schema_id).await?;
+
+    VariantAuthoringClient::save_variant_content(
+        &ctx,
+        schema_variant_id,
+        &variant.schema_name,
+        variant.display_name.clone(),
+        variant.category.clone(),
+        variant.description.clone(),
+        variant.link.clone(),
+        variant.color.clone(),
+        variant.component_type.into(),
+        Some(code),
+    )
+    .await?;
```

Comment:
Love the minimal route. Is it possible to get the asset sprayer into an integration test? I suspect the required token makes that difficult, though I wonder if we can test the library without generating real code.